platform/security/encryption/platform-security-encryption-impl/src/main/java/ddf/security/encryption/impl/EncryptionServiceImpl.java

```diff
@@ -28,54 +42,75 @@
 
   private static final Pattern ENC_PATTERN = Pattern.compile("^ENC\\((.*)\\)$");
   private static final String ENC_TEMPLATE = "ENC(%s)";
+  private static final String KEYSET_FILE_NAME = "keyset.json";
+  private static final byte[] ASSOCIATED_DATA = "associated.data".getBytes();
 
-  private final Crypter crypter;
+  @VisibleForTesting KeysetHandle keysetHandle;
+  private Aead aead;
 
   public EncryptionServiceImpl() {
     String passwordDirectory = System.getProperty("ddf.etc").concat("/certs");
+    String keysetLocation = Paths.get(passwordDirectory, KEYSET_FILE_NAME).toString();
 
     synchronized (EncryptionServiceImpl.class) {
-      if (!new File(passwordDirectory.concat("/meta")).exists()) {
-        KeyczarTool.main(
-            new String[] {
-              "create", "--location=" + passwordDirectory, "--purpose=crypt", "--name=Password"
-            });
-        KeyczarTool.main(
-            new String[] {"addkey", "--location=" + passwordDirectory, "--status=primary"});
-      }
-      Crypter crypter = null;
+      File keysetFile = new File(keysetLocation);
+      InputStream keysetFileInputStream = null;
+      OutputStream keysetFileOutputStream = null;
       try {
-        crypter = new Crypter(passwordDirectory);
-      } catch (Exception e) {
-        LOGGER.debug(e.getMessage());
+        AeadConfig.register();
+        if (!keysetFile.exists()) {
+          keysetHandle = KeysetHandle.generateNew(AeadKeyTemplates.AES128_GCM);
+          keysetFileOutputStream = Files.newOutputStream(Paths.get(keysetLocation));
+          CleartextKeysetHandle.write(
+              keysetHandle, JsonKeysetWriter.withOutputStream(keysetFileOutputStream));
+        } else {
+          keysetFileInputStream = Files.newInputStream(Paths.get(keysetLocation));
+          keysetHandle =
+              CleartextKeysetHandle.read(JsonKeysetReader.withInputStream(keysetFileInputStream));
+        }
+        aead = AeadFactory.getPrimitive(keysetHandle);
+      } catch (GeneralSecurityException | IOException e) {
+        LOGGER.warn("Problem initializing Tink. Enable debug logging for more information.");
+        LOGGER.debug("", e);
+      } finally {
+        // close streams
+        try {
+          keysetFileInputStream.close();
+        } catch (IOException | NullPointerException ignore) {
```

Comment:
✏️ Do a null check instead of catching the NPE. Can a try with resources be used?